src/components/EditForm.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState } from 'react';
+import type { FormEvent } from 'react';
+import type { TodoItem } from '@/types/todo.types';
+
+interface EditTodoFormProps {
+  editTodo: (updatedTask: string, id: string) => void;
+  task: TodoItem;
+}
+
+export const EditTodoForm: React.FC<EditTodoFormProps> = ({ editTodo, task }) => {
+  const [value, setValue] = useState<string>(task.task);
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (!value.trim()) return;
```

Comment:
Nit: Check if the trimmed value is actually different from the original before calling `onEditTodo`.

```suggestion
    if (!value.trim() || value.trim() === task.task.trim()) return;
```

Reply:
add https://github.com/Sakshi-Simform/todo-list/pull/3/commits/fa7ca6370419105705ca120a892bfcb1180f21e6